### PR TITLE
net: promiscuous: Fix crash in promiscuous mode

### PR DESCRIPTION
**Pull request by icsys-aal**
When a packet can't be cloned we crash as we try to initialize the cursor on a nullptr. We should check if we have a valid pointer, and if we don't we drop the packet along with a warning.

### CODE REVIEW

**Review by jukkar — COMMENTED** [review]
The change is good, thanks for fixing it. Just fix the compliance issues and we are ready to merge this.

**Comment by icsys-aal** [comment]
Fixed the code formatting, let me know if there's anything else you need me to change.

File: subsys/net/ip/net_if.c [hunk 1]
```diff
@@ -4145,12 +4145,16 @@ enum net_verdict net_if_recv_data(struct net_if *iface, struct net_pkt *pkt)
...
+			net_pkt_cursor_init(new_pkt);
 
-		if (net_promisc_mode_input(new_pkt) == NET_DROP) {
-			net_pkt_unref(new_pkt);
+			if (net_promisc_mode_input(new_pkt) == NET_DROP) {
+				net_pkt_unref(new_pkt);
+			}
+		} else {
+			NET_WARN("packet dropped, unable to clone packet");
+			verdict = NET_DROP;
```
Comment (jukkar) [inline subsys/net/ip/net_if.c:4156]: After revisiting this change, I am not sure about this `NET_DROP`. The reason being that the function should return info whether the original net_pkt should be dropped or not, and in this case the earlier call
`verdict = net_if_l2(iface)->recv(iface, pkt);` returns that information.
So for the cloned pkt, imho we should not set the verdict here. To me it looks like that if we return `NET_DROP` here, the original pkt is dropped without any reason. WDYT?

Comment (jukkar) [reply subsys/net/ip/net_if.c:4156]: Also in the case of the cloning failed, perhaps better error msg would be
`NET_WARN("promiscuous packet dropped, unable to clone packet");`
in order to tell the reader what exactly happened.

**Comment by jukkar** [comment]
There is still mismatch for your signed-off-by, please see
https://github.com/zephyrproject-rtos/zephyr/actions/runs/6183457730/job/16787172023?pr=62582

File: subsys/net/ip/net_if.c (hunk 1 above)
Comment (icsys-aal) [reply subsys/net/ip/net_if.c]: After looking into it I agree with your suggestion and have removed the verdict part.

**Review by jukkar — APPROVED** [review]

**Comment by jukkar** [comment]
@icsys-aal please check the compliance issues, the checker is very picky about your commit author name, it should match the signed-off-by

**Review by jukkar — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**